db.router.js

```diff
@@ -0,0 +1,85 @@
+const express = require('express');
+const db = require('./data/db');
+
+const router = express.Router();
+
+
+//GET request
+router.get('/', async(req, res) => {
+    try{
+        const users = await db.find(req.query);
+        res.status(200).json(users)
+    } catch(err) {
+        res.status(500).json({error: "The posts information could not be retrieved."})
+    }
+})
+
+//GET by id
+router.get('/:id', async(req, res) => {
+    try{
+        const user = await db.findById(req.params.id)
+        if(user){
+            res.status(200).json(user)
+        } else {
+            res.status(404).json({ message: "The post with the specified ID does not exist." })
+        }
+    } catch {
+        res.status(500).json({
+            message:'Error retrieving the user'
+        })
+    }
+})
+
+//POST 
+router.post('/', async (req, res) => {
+    const newUser = req.body
+    const {title, contents} = req.body;
+
+    if( title && contents) {
```

Comment:
you may want to instead check if these values are empty. I'm pretty sure title is notNullable() which means it requires an input. This current setup will allow empty strings to pass through with no problem. You don't want a database full of empty strings.